### PR TITLE
server/new: optional secure-cookie/HSTS security.

### PR DESCRIPTION
**Pull request by clint-tseng**
* change the cookie_ssl_only setting to true to enable.
* this will turn on:
  * secure cookies, meaning HTTPS cookies cannot be read from HTTP.
  * HSTS with max-age of 90 days, indicating that for at least 90 days
    the site may only be accessed via HTTPS and never from HTTP.

### CODE REVIEW

File: tls_odkbuild.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      @app = app
+    end
+
+    def call(env)
+      @app.call(env).tap do |response|
+        # grab headers
+        headers = response[1]
+
+        # add HSTS
+        headers['Strict-Transport-Security'] = 'max-age=778000'
```
Comment (lognaturel) [inline tls_odkbuild.rb]: This is actually about 9 days. Should it be 90?

File: config.yml.sample [hunk 2]
```diff
@@ -1,5 +1,6 @@
 development:
   cookie_secret: sample
+  cookie_ssl_only: false
```
Comment (lognaturel) [inline config.yml.sample]: Why are they off by default?

Comment (clint-tseng) [inline config.yml.sample]: For one, it would break your favored default Vagrant deployment.

File: tls_odkbuild.rb (hunk 1 above)
Comment (clint-tseng) [inline tls_odkbuild.rb]: that's what i get for believing people on the internet.

File: config.yml.sample (hunk 2 above)
Comment (lognaturel) [inline config.yml.sample]: Ok, any other reasons? I'm not too familiar with the implications one way or the other.

Comment (clint-tseng) [inline config.yml.sample]: nope. that's the main one; `vagrant up` would not result in a functional VM.

(no public DNS means no ability to procure a letsencrypt cert; no cert means nginx will balk at spinning up the SSL server; balking at spinning up the SSL server means nginx will not start at all; nginx not starting means ansible will not complete execution.)

Comment (lognaturel) [inline config.yml.sample]: Ok, great. Then I think leave this as it is and in the README you can list this as a reason someone might want to do manual provisioning.

File: tls_odkbuild.rb (hunk 1 above)
Comment (lognaturel) [inline tls_odkbuild.rb]: I am ready to merge once you update this.

**Pull request merged**